src/server/hotkey_collector.cpp

```diff
@@ -17,21 +17,38 @@
 
 #include "hotkey_collector.h"
 
+#include <dsn/dist/replication/replication_enums.h>
 #include <dsn/utility/smart_pointers.h>
 #include "base/pegasus_key_schema.h"
+#include <dsn/dist/fmt_logging.h>
 
 namespace pegasus {
 namespace server {
 
-hotkey_collector::hotkey_collector()
-    : _internal_collector(dsn::make_unique<hotkey_empty_data_collector>())
+hotkey_collector::hotkey_collector(dsn::replication::hotkey_type::type hotkey_type,
+                                   dsn::replication::replica_base *r_base)
+    : replica_base(r_base),
+      _state(hotkey_collector_state::STOPPED),
+      _hotkey_type(hotkey_type),
+      _internal_collector(std::make_shared<hotkey_empty_data_collector>())
 {
 }
 
-// TODO: (Tangyanzhao) implement these functions
 void hotkey_collector::handle_rpc(const dsn::replication::detect_hotkey_request &req,
                                   dsn::replication::detect_hotkey_response &resp)
 {
+    switch (req.action) {
+    case dsn::replication::detect_action::START:
+        on_start_detect(resp);
+        return;
+    case dsn::replication::detect_action::STOP:
+        on_stop_detect(resp);
+        return;
+    default:
+        resp.err = dsn::ERR_INVALID_VERSION;
+        resp.__set_err_hint("can't find this detect_action");
+        dwarn_replica("can't find this detect_action");
```

Comment:
Add error details, i.e. `req.action`, into hint message and log, then admins know what happend.

Reply:
if `req.action` is an unknown enum, I can't transform it into a string to log it.